client/asset/btc/spv_wrapper.go

```diff
@@ -158,10 +165,25 @@ func (s *btcChainService) Peers() []SPVPeer {
 	return peers
 }
 
+func (s *btcChainService) AddPeer(addr string) error {
+	serverPeer := neutrino.NewServerPeer(s.ChainService, true)
+	peer, err := peer.NewOutboundPeer(neutrino.NewPeerConfig(serverPeer), addr)
+	if err != nil {
+		return err
+	}
+	conn, err := net.Dial("tcp", peer.Addr())
+	if err != nil {
+		return err
+	}
+	peer.AssociateConnection(conn)
+	serverPeer.Peer = peer
+	return nil
```

Comment:
I see this part repeated in **bch** and **ltc**. Perhaps an exported helper function that accepts the proper interfaces would be usefuil?

Reply:
All the functions on these `spvService` types are completely repeated. There are so many different types used in these functions from the three neutrino libraries that are all the same, I feel like it's simpler just to repeat them. I can try to do something if you feel strongly about it though.

Reply:
Oh yeah. It'd be messy.